wasm-tools: load new tag format

ydnar opened the pull request:
Fixes #8.

- Loads from legacy and new tag formats
- For consistency, all setup scripts now accept a version spec of `v1.0.0` or `1.0.0`
- I renamed CI test job names for readability in the GitHub UI

ydnar committed:
wasm-tools: load from correct tag

Fixes bytecodealliance/actions#8.

wasm-tools: try legacy tag format, then new tag format; strip v from version number

.github/workflows/wasmtools-tests: add tests for 1.200.0 and vN

wasmtools -> wasm-tools

To match the crate and executable name.

.github/workflows/wasm-tools-tests: fix detection

wasm-tools: fix lint issues

wasmtime: accept version values prefixed with 'v'

wit-bindgen: support new tag format, versions with v

.github/workflows: fix failing tests

wit-bindgen: that's what I get for copy and paste early in the morning

wasmtime: correct tag format

wasmtime, wit-bindgen: version string used in copy

wit-bindgen: mutate version variable depending on tag format

wit-bindgen: try this

wit-bindgen: try separate vVersion variable

wit-bindgen: try this

wit-bindgen: use separate binVersion

.github/workflows: rename build jobs with shorter names

So the full matrix names can be seen in GitHub Actions runner sidebar

.github/workflows: move version in front of os for display purposes

.github/workflows: break out tests for v-prefixed versions

rajatjindal commented:
thank you so much for the PR (and all the work you do for open source community).

if you have a few mins, could you please add a testcase in https://github.com/bytecodealliance/actions/blob/main/.github/workflows/wasmtools-tests.yml for the new wasm-tools format. (or I am happy to add it myself in a follow up PR).

>  I’m happy to update this to have wasmtime accept a bare version without the v prefix.

what do you think would be better: to accept version without v prefix or update actions to support both formats (with/without prefix) to avoid any confusions.

thanks

ydnar commented:
Thanks for making this! I have a few projects that use it—it’s been handy to have an official setup action for `wasmtime`, `wasm-tools`, and `wit-bindgen`.

ydnar commented:
OK, tests pass. I rebased and squashed.

ydnar commented:
Would it make sense to bump the release to `v1.1` after this is merged?

rajatjindal commented:
Hi @ydnar , thank you for making the changes and adding the tests. 

given these changes are backward compatible (please correct me I misread this), I think we can just make a patch release. 

In anycase, for `v1.x.y`, we will move tag `v1` to point to latest release automatically (this is the norm I have seen in other actions), so it should be fine either way.

Thanks

rajatjindal commented:
Please let me know if you have opinion about patch vs minor release, and I will cut the new release accordingly. thanks again.

ydnar commented:
It's backwards compatible as far as I can tell from the tests. Any version is probably fine. 

Do you want to update the README?

rajatjindal commented:
yeah, that would be great. thank you.

ydnar committed:
README: updates

- Add wit-bindgen
- Change wasmtime version argument to omit the v prefix
- Markdown lint/format/remove trailing whitespace
- Add backticks around names

ydnar commented:
Done!

ydnar committed:
README: further tweaks
